### PR TITLE
feat(functions) add tribe_get_query_var function

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: http://central.tri.be/issues/137262

This PR adds the `tribe_get_query_var` function, then used in https://github.com/moderntribe/events-pro/pull/1343

### CODE REVIEW

File: src/functions/utils.php [hunk 1]
```diff
@@ -717,3 +717,33 @@ function tribe_sanitize_deep( &$value ) {
...
+
+		$query = wp_parse_url( $url, PHP_URL_QUERY );
+
+		if ( empty( $query ) ) {
+			return $default;
+		}
+
+		wp_parse_str( $query, $parsed );
+
+		return ! empty( $parsed ) && isset( $parsed[ $query_arg ] ) ? $parsed[ $query_arg ] : $default;
```
Comment (bordoni) [inline src/functions/utils.php]: Lets use `Arr::get()` here.

File: tests/wpunit/Tribe/functions/utilsTest.php [hunk 2]
```diff
@@ -218,4 +218,24 @@ public function tribe_sanitize_deep_data_set() {
...
+	public function tribe_get_query_var_data_set() {
+		return [
+			'empty'                 => [ '', 'test', null ],
+			'not_a_url'             => [ 'foo-bar-baz', 'test', null ],
+			'abs_no_query'          => [ 'http://example.com/', 'test', null ],
+			'rel_no_query'          => [ '/index.php', 'test', null ],
+			'abs_query_arg_not_set' => [ 'http://example.com/?foo=bar', 'test', null ],
+			'rel_query_arg_not_set' => [ '/index.php?foo=bar', 'test', null ],
+			'abs_query_arg_set'     => [ 'http://example.com/?test=bar', 'test', 'bar' ],
+			'rel_query_arg_set'     => [ '/index.php?test=bar', 'test', 'bar' ],
```
Comment (bordoni) [inline tests/wpunit/Tribe/functions/utilsTest.php]: We should add a test to passing an array in the query args.

Comment (lucatume) [reply tests/wpunit/Tribe/functions/utilsTest.php]: Do we want to support arrays?

Comment (lucatume) [reply tests/wpunit/Tribe/functions/utilsTest.php]: Done

File: src/functions/utils.php [hunk 3]
```diff
@@ -717,3 +717,42 @@ function tribe_sanitize_deep( &$value ) {
...
+
+		$query = wp_parse_url( $url, PHP_URL_QUERY );
+
+		if ( empty( $query ) ) {
+			return $default;
+		}
+
+		wp_parse_str( $query, $parsed );
+
+		if (!is_array($query_arg)) {
```
Comment (bordoni) [inline src/functions/utils.php]: spacing.

**Pull request merged**